plugins/toolkit/jetbrains-core/src/software/aws/toolkits/jetbrains/services/codewhisperer/util/CodeWhispererEndpointCustomizer.kt

```diff
@@ -56,6 +64,41 @@ class CodeWhispererEndpointCustomizer : ToolkitClientCustomizer {
                     }
                 }
             )
+            if (builder is CodeWhispererStreamingAsyncClientBuilder) {
+                val proxy = CommonProxy.getInstance().select(endpoint).first()
+                val address = proxy.address()
+                val clientBuilder = NettyNioAsyncHttpClient.builder()
+
+                // proxy.type is one of {DIRECT, HTTP, SOCKS}, and is definitely a InetSocketAddress in the HTTP/SOCKS case
+                // and is null in DIRECT case
+                if (proxy.type() == Proxy.Type.SOCKS) {
+                    error("Q Chat HTTP client does not support SOCKS proxies")
+                } else if (address is java.net.InetSocketAddress) {
+                    val proxyConfiguration = ProxyConfiguration.builder()
+                        .host(address.getHostName())
+                        .port(address.getPort())
+                        .apply {
+                            val configurable = HttpConfigurable.getInstance()
+                            val proxyExceptions = configurable.PROXY_EXCEPTIONS
+                            if (!proxyExceptions.isNullOrBlank()) {
+                                // should be handled by CommonProxy, but also should be no harm in passing this along if something more complicated is happening
+                                nonProxyHosts(proxyExceptions.split(',').toSet())
+                            }
+
+                            val login = HttpConfigurable.getInstance().proxyLogin
+                            if (login != null) {
+                                username(login)
+                                password(HttpConfigurable.getInstance().plainProxyPassword)
+                            }
```

Comment:
i dont think this works if the user prefers to enter in credentials interactively, instead of saving them in the configuration